tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -0,0 +1,233 @@
+    trigger Vulnerability Detector scans, generating the expected vulnerabilities.
+    The verification of vulnerabilities is conducted through the Wazuh Indexer, Agents' state index, and Wazuh API vulnerability endpoints.
+    Additionally, the tests ensure the consistency of these values.
+
+Tests:
+    - test_syscollector_initial_agent_scan: 
+        Validates the initiation of Syscollector scans across all agents in the environment. 
+        Subsequently, it ensures that the Vulnerability Detector accurately detects vulnerabilities within the environment. 
+        The Agent's Vulnerability Indexer index is expected to be updated with the detected vulnerabilities.
+
+Issue: https://github.com/wazuh/wazuh-qa/issues/4369
+
+Components:
+  - vulnerability_detector
+  - syscollector
+
+Environment: e2e_vulnerability_detector
+
+Tiers: 0
+
+Markers:
+  - e2e
+  - vulnerability_detector
+  - tier0
+
+ToDo:
+    - check_vuln_state_index: Function to ensure the consitency of the agent's vulnerabilities
+    - Check if wait_until_vuln_scan_agents_finished function works as expected for the second scan
+"""
+import os
+import pytest
+import logging
+
+from typing import Generator, Dict, List
+
+from wazuh_testing.tools.configuration import load_configuration_template
+from wazuh_testing.end_to_end.configuration import backup_configurations, restore_backup, configure_environment
+from wazuh_testing.end_to_end.logs import truncate_agents_logs, truncate_managers_logs, truncate_logs
+from wazuh_testing.end_to_end.wazuh_api import  get_agents_vulnerabilities
+from wazuh_testing.end_to_end.waiters import wait_until_vd_is_updated, wait_until_vuln_scan_agents_finished
+from wazuh_testing.end_to_end.monitoring import generate_monitoring_logs_all_agent, monitoring_events_multihost
+from wazuh_testing.end_to_end.regex import get_event_regex
+from wazuh_testing.end_to_end.indexer_api import get_indexer_values
+from wazuh_testing.tools.system import HostManager
+
+
+TIMEOUT_SYSCOLLECTOR_SCAN = 200
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+local_path = os.path.dirname(os.path.abspath(__file__))
+current_dir = os.path.dirname(__file__)
+configurations_dir = os.path.join(current_dir, "data", "configurations")
+configurations_paths = {
+        'manager': os.path.join(configurations_dir, 'manager.yaml'),
+        'agent': os.path.join(configurations_dir, 'agent.yaml')
+        }
+
+
+def load_vulnerability_detector_configurations():
+    """Return the configurations for Vulnerability testing for the agent and manager roles
+
+    Return:
+        Dict: Configurations for each role
+    """
+    return {
+        'agent': load_configuration_template(configurations_paths['agent'], [{}], [{}]),
+        'manager': load_configuration_template(configurations_paths['manager'], [{}], [{}])
+    }
+
+
+@pytest.fixture(scope='module')
+def setup_vulnerability_tests(host_manager: HostManager) -> Generator:
```

Comment:
Check logger

Reply:
Could you please extend this?

Reply:
In https://github.com/wazuh/wazuh-qa/issues/4591 we will replace this for log function in order to avoid confussions